src/mpid/common/shm/mpidu_init_shm_alloc.c

```diff
@@ -220,19 +220,17 @@ static int check_alloc(MPIDU_shm_seg_t * memory, int local_rank)
 
     if (MPIR_Process.local_rank == 0) {
         asym_check_region_p->base_ptr = memory->base_addr;
-        OPA_store_int(&asym_check_region_p->is_asym, 0);
+        MPL_atomic_store_int(&asym_check_region_p->is_asym, 0);
     }
 
     MPIDU_Init_shm_barrier();
 
     if (asym_check_region_p->base_ptr != memory->base_addr)
-        OPA_store_int(&asym_check_region_p->is_asym, 1);
-
-    OPA_read_write_barrier();
+        MPL_atomic_store_int(&asym_check_region_p->is_asym, 1);
 
     MPIDU_Init_shm_barrier();
 
-    if (OPA_load_int(&asym_check_region_p->is_asym)) {
+    if (MPL_atomic_load_int(&asym_check_region_p->is_asym)) {
         memory->symmetrical = 0;
     } else {
         memory->symmetrical = 1;
```

Comment:
Actually, this can be removed following this commit: https://github.com/pmodels/mpich/pull/3916/commits/43043ed29894fa5092993ca4b930960ea4de5385

I didn't merged this because the PR it was part of got dropped. Should I create a PR for it separately?

Reply:
It might be easier I just cherry pick that commit